catroid/src/androidTest/java/org/catrobat/catroid/test/io/DefaultDroneProgramLoadingTest.java

```diff
@@ -0,0 +1,113 @@
+import org.catrobat.catroid.exceptions.LoadingProjectException;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import static junit.framework.Assert.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+public class DefaultDroneProgramLoadingTest {
+	private final XstreamSerializer storageHandler;
+	private String projectName;
+	private Project currentProjectBuffer;
+	private Context targetContext;
+	private Project project;
+
+	public DefaultDroneProgramLoadingTest() {
+		storageHandler = XstreamSerializer.getInstance();
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		targetContext = InstrumentationRegistry.getTargetContext();
+		currentProjectBuffer = ProjectManager.getInstance().getCurrentProject();
+		projectName = targetContext.getString(R.string.default_drone_project_name);
+		project = new ArDroneProjectCreator().createDefaultProject(projectName, targetContext, false);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		ProjectManager.getInstance().setProject(currentProjectBuffer);
+		TestUtils.deleteProjects(projectName);
+	}
+
+	@Test
+	public void testDroneProgramLoadingSuccessfully() throws IOException, LoadingProjectException {
+		Project loadedProject = storageHandler.loadProject(projectName, targetContext);
+		Scene preScene = project.getDefaultScene();
+		Scene postScene = loadedProject.getDefaultScene();
+
+		ArrayList<Sprite> preSpriteList = (ArrayList<Sprite>) project.getDefaultScene().getSpriteList();
+		ArrayList<Sprite> postSpriteList = (ArrayList<Sprite>) loadedProject.getDefaultScene().getSpriteList();
+
+		assertEquals(project.getName(), loadedProject.getName());
+		assertEquals(preScene.getName(), postScene.getName());
+
+		assertEquals(preSpriteList.get(0).getName(), postSpriteList.get(0).getName());
+		assertEquals(preSpriteList.get(1).getName(), postSpriteList.get(1).getName());
+		assertEquals(preSpriteList.get(2).getName(), postSpriteList.get(2).getName());
+		assertEquals(preSpriteList.get(3).getName(), postSpriteList.get(3).getName());
+		assertEquals(preSpriteList.get(4).getName(), postSpriteList.get(4).getName());
+		assertEquals(preSpriteList.get(5).getName(), postSpriteList.get(5).getName());
+		assertEquals(preSpriteList.get(6).getName(), postSpriteList.get(6).getName());
+		assertEquals(preSpriteList.get(7).getName(), postSpriteList.get(7).getName());
+		assertEquals(preSpriteList.get(8).getName(), postSpriteList.get(8).getName());
+		assertEquals(preSpriteList.get(9).getName(), postSpriteList.get(9).getName());
+		assertEquals(preSpriteList.get(10).getName(), postSpriteList.get(10).getName());
+		assertEquals(preSpriteList.get(11).getName(), postSpriteList.get(11).getName());
+
+		assertEquals(1, postSpriteList.get(0).getLookList().size());
```

Comment:
the 1s for the expected value would probably be nice as a constant e.g. expectedListSize

just a bit better for readability imo

Reply:
ok. done